### DISCUSSION TITLE
How to add custom route to all k3s nodes?

### DISCUSSION THREAD

**Discussion by ralimardanov** [mysticaltech/terraform-hcloud-kube-hetzner#1014] [Q&A]
Hello.
First of all, I want to thank you for such nice module! It helped us a lot.

I have a question, so we have configured infra with your module and made site-to-site vpn with other client.
To reach clients network, traffic should go through VPN.
For containers running inside cluster, we have to add 1 route in Hetzner network, but because it's external subnet, it asks us to manually add route on all nodes:
<img width="387" alt="image" src="https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/38042242/2ba78cc3-5c67-4418-82f7-d4ff3269f053">
When adding manually route to one of our nodes, connection works. The thing is we wanted to have this automatically, especially for autoscaling nodes. I tried to search a bit about how servers are created and if there is any way to pass such configuration, but with no luck.
@mysticaltech or anyone else who has faced such situation, what can you suggest?
Thanks in advance!

PS: I don't know if kube.tf file is needed for this case, so I didn't add it right now.

**Comment by mysticaltech** [comment]
@ralimardanov That's certainly possible, but we do not explicitely have that feature that this moment. It would need to be added, For instance you could add a variable with an array of routes to add to each node, this would be done in the internal host module (see module/host) most probably, as it's the part in common to all nodes.

@M4t7e FYI, maybe you'd have some ideas for that too.

**Reply by ralimardanov** [reply]
Hello @mysticaltech , thanks for your fast reply.
I tried to add something like this in cloudinit.yaml.tpl in module/host templates:
> network:
>   version: 1
>   config:
>     - type: physical
>       name: eth1
>       subnets:
>         - type: static
>           address: <destination_subnet>
>           gateway: <gateway_address>

But it didn't seem to work out. Do you have anything on mind about how it should be done?

**Reply by mysticaltech** [reply]
@ralimardanov We already add a route somewhere, just search for that and do the same thing.
